ctru-rs/src/lib.rs

```diff
@@ -11,20 +11,27 @@ pub fn init() {
 
     use std::panic::PanicInfo;
 
+    let main_thread = thread::current().id();
+
     // Panic Hook setup
     let default_hook = std::panic::take_hook();
     let new_hook = Box::new(move |info: &PanicInfo| {
-        println!("\x1b[1;31m\n--------------------------------------------------");
+        println!("\x1b[1;31m\n--------------------------------------------------"); // Red ANSI Color Code
         default_hook(info);
-        println!("\nPress SELECT to exit the software");
-        let hid = services::hid::Hid::init().unwrap();
 
-        loop {
-            hid.scan_input();
-            if hid.keys_down().contains(services::hid::KeyPad::KEY_SELECT) {
-                break;
+        // Only for panics in the main thread
+        if main_thread == thread::current().id() {
+            println!("\nPress SELECT to exit the software");
+            let hid = services::hid::Hid::init().unwrap();
+
+            loop {
+                hid.scan_input();
+                if hid.keys_down().contains(services::hid::KeyPad::KEY_SELECT) {
+                    break;
+                }
             }
         }
+        println!("\x1b[0m"); // Get back to white
```

Comment:
White might not be the color they were using

Reply:
I would expect the color to be specified each time, though it doesn't solve the issue. Can we make "one-time" ANSI codes? I don't know.

P.S.
If a thread crashes, you have bigger problems than using the wrong color.

Reply:
Since ANSI is parsed as normal text, and that text (in our case) ends up in `stdout`, we can't do much about missing colors. Usually speaking, it is indeed expected from the user to use codes for each colored text, but the `Console`s don't reset the color each time, so I guess we must make a choice:

We leave it this way and leave the burden on the user *or* we don't use red, though a green error text would be jarring.

We must also consider the user is already expected to have their console screwed up one way or another, since printing with `newline`s usually make the whole screen's text wrap. Tell me what you think

Reply:
As an experiment, I ran this code on my host system:
```rust
fn main() {
    let red = ansi_term::Color::Red;
    eprintln!("{}hello world", red.prefix());

    panic!("main thread panic");
}
```

The panic output is indeed colored red: 
![Screen Shot 2022-01-27 at 3 36 06 PM](https://user-images.githubusercontent.com/11131775/151440671-1865b4e2-eb87-4daa-9e57-1b8ba5b0ee5a.png)

There is no special handling to "uncolor" text when printing panics to stderr on a typical OS, so one could argue that maybe there shouldn't be any code to color it either? `"\x1b[0m"` technically resets it back to unstyled text, which might _also_ not be what the user had before panicking. 

Since the console is global, any thread that prints to it could modify the colors using ANSI codes, so I'm not sure it makes sense to try and work around that...

Reply:
> Since the console is global, any thread that prints to it could modify the colors using ANSI codes, so I'm not sure it makes sense to try and work around that...

Basically this, we can either choose to *not* add to the chaos caused by the codes, or adding to it expecting some handling to exist anyways since the chaos itself exists anyways. Both approaches are equally valid. 

@AzureMarker @ian-h-chamberlain tell me what you prefer, a vote works best with an uneven number of voters.

Reply:
I think my vote is to remove color / special formatting altogether from the default hook and let users add it if they want. Some applications may prefer to panic to the bottom screen, or use wide mode, etc., and I think color formatting falls into the same category. In any of those use cases someone could always do this or equivalent in their app setup to achieve the same thing, right?

```rust
std::panic::set_hook(Box::new(move |info| {
    let red = ansi_term::Color::Red;
    eprint!("{}", red.prefix());
    old_hook(info);
    eprint!("{}", red.suffix());
}));
```

Reply:
I'm also for this. No red coloring and no weird stuff. If it crashes, something is print, that's it.

Would like to hear from @AzureMarker for suggestions.

Reply:
Yeah removing the coloring from the ctru hook sounds good to me.